raiden-ts/src/transport/epics.ts

```diff
@@ -1503,3 +1520,352 @@ export const deliveredEpic = (
     }),
   );
 };
+
+type CallInfo = { callId: string; peerId: string; peerAddress: Address };
+
+// setup candidates$ handlers
+function handleCandidates$(
+  connection: RTCPeerConnection,
+  matrix: MatrixClient,
+  start$: Observable<null>,
+  { callId, peerId, peerAddress }: CallInfo,
+  { log, latest$, config$ }: Pick<RaidenEpicDeps, 'log' | 'latest$' | 'config$'>,
+) {
+  return merge(
+    // when seeing an icecandidate, send it to peer
+    fromEvent<RTCPeerConnectionIceEvent>(connection, 'icecandidate').pipe(
+      pluck('candidate'),
+      delayWhen(() => start$),
+      takeWhile(isntNil),
+      bufferTime(10),
+      filter((candidates) => candidates.length > 0),
+      tap((e) => log.debug('RTC: got candidates', callId, e)),
+      mergeMap((candidates) =>
+        waitMemberAndSend$(
+          peerAddress,
+          matrix,
+          'm.call.candidates' as EventType,
+          { call_id: callId, version: 0, candidates },
+          { log, latest$, config$ },
+        ),
+      ),
+    ),
+    // when receiving candidates from peer, add it locally
+    fromEvent<MatrixEvent>(matrix, 'event').pipe(
+      filter(
+        (event) =>
+          event.getType() === ('m.call.candidates' as EventType) &&
+          event.getSender() === peerId &&
+          event.event?.content?.call_id === callId,
+      ),
+      tap((e) => log.debug('RTC: received candidates', callId, e.event?.content?.candidates)),
+      mergeMap((event) => from<RTCIceCandidateInit[]>(event.event?.content?.candidates ?? [])),
+      mergeMap((candidate) =>
+        defer(() => connection.addIceCandidate(new RTCIceCandidate(candidate))).pipe(
+          catchError((err) => {
+            log.error('RTC: error setting candidate, ignoring', err);
+            return EMPTY;
+          }),
+        ),
+      ),
+    ),
+  ).pipe(ignoreElements());
+}
+
+// setup channel for caller
+function setupCallerChannel$(
+  matrix: MatrixClient,
+  start$: Subject<null>,
+  info: CallInfo,
+  { httpTimeout }: RaidenConfig,
+  deps: Pick<RaidenEpicDeps, 'log' | 'latest$' | 'config$'>,
+) {
+  const { callId, peerId, peerAddress } = info;
+  const { log, latest$, config$ } = deps;
+
+  const connection = new RTCPeerConnection({
+    iceServers: [{ urls: 'stun:stun.l.google.com:19302' }],
```

Comment:
Maybe move to `RaidenConfig`

Reply:
Yes, it's partly in that firs TODO in this issue, the default/fallback server needs to be configured/defined somewhere else, and we should be able to use the turn servers provided by matrix, if any; thanks for the headsup.

Reply:
How would Matrix provide these? I wasn't able to find it in their docs. They already do this for VoIP right?

Reply:
To setup: https://github.com/matrix-org/synapse/blob/master/docs/turn-howto.md
To fetch on clients:
https://matrix.org/docs/spec/client_server/latest#get-matrix-client-r0-voip-turnserver
https://github.com/matrix-org/matrix-js-sdk/blob/1834e6688fc819e0d6d45306faa710b2db374997/src/client.js#L4589-L4591

Reply:
Done, set in `fallbackIceServers` in config, and also fetching (with cache) from matrix, if available. Thanks